python/dgl/optim/pytorch/sparse_optim.py

```diff
@@ -25,6 +25,35 @@ def __init__(self, params, lr):
         self._world_size = None
         self._shared_cache = {}
         self._clean_grad = False
+        self._opt_meta = {}
+
+        for emb in params:
+            assert isinstance(emb, NodeEmbedding), \
+                'DGL SparseOptimizer only supports dgl.nn.NodeEmbedding'
+
+            if self._rank is None:
+                self._rank = emb.rank
+                self._world_size = emb.world_size
+            else:
+                assert self._rank == emb.rank, \
+                    'MultiGPU rank for each embedding should be same.'
+                assert self._world_size == emb.world_size, \
+                    'MultiGPU world_size for each embedding should be same.'
+
+            emb_name = emb.name
+            if self._rank <= 0 and self._world_size > 1:
+                opt_meta = create_shared_mem_array(emb_name+'_opt_meta', \
+                    (self._world_size, self._world_size), th.int32).zero_()
+            if self._rank == 0:
+                if self._world_size > 1:
```

Comment:
how is this condition different from the one above (line 44)?

Reply:
For single gpu training, rank is also 0

Reply:
but then you have `if self._world_size > 1`. Is the condition the same as above?